tests/js/client/http/coordinator-metrics-cluster.js

```diff
@@ -0,0 +1,128 @@
+      let foo2_values = [{"name": "i"}];
+      let foo3_values = [{"name": "hate"}, {"name": "js"}];
+      for (let i = 0; i !== 100; ++i) {
+        foo1_values.push({"name": i});
+        foo2_values.push({"name": i + 100000});
+        foo3_values.push({"name": i - 100000});
+      }
+      db.foo1.save(foo1_values);
+      db.foo2.save(foo2_values);
+      db.foo3.save(foo3_values);
+      db._createView("foov", "arangosearch", {
+        "links": {
+          "foo1": {"includeAllFields": true},
+          "foo2": {"includeAllFields": true},
+          "foo3": {"includeAllFields": true}
+        }
+      });
+      db._query("FOR d IN foov OPTIONS { waitForSync: true } LIMIT 1 RETURN 1");
+    },
+
+    tearDownAll: function () {
+      db._dropView("foov");
+      db._drop("foo1");
+      db._drop("foo2");
+      db._drop("foo3");
+    },
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief CoordinatorMetricsTestSuite tests
+////////////////////////////////////////////////////////////////////////////////
+
+    testIResearchLinkStats: function () {
+      let coordinators = getEndpointsByType("coordinator");
+      let c = coordinators[0];
+      {// make metrics actual
+        getMetricRaw(c);
+        require('internal').sleep(2);
+      }
+      let txt = getMetricRaw(c);
+      let json = parsePrometheusTextFormat(txt);
+      let metrics = {};
+      json.forEach((entry) => {
+        entry.metrics.forEach((entry2) => {
+          if (!_.has(entry2, "labels") || !_.has(entry2.labels, "collection")) {
+            return;
+          }
+          if (!_.has(metrics, entry.name)) {
+            metrics[entry.name] = {};
+          }
+          metrics[entry.name][entry2.labels.collection] = entry2.value;
+        });
+      });
+
+      assertEqual(metrics["arangodb_search_num_docs"]["foo1"], 100);
+      assertEqual(metrics["arangodb_search_num_docs"]["foo2"], 101);
+      assertEqual(metrics["arangodb_search_num_docs"]["foo3"], 102);
+
+      assertEqual(metrics["arangodb_search_num_live_docs"]["foo1"], 100);
+      assertEqual(metrics["arangodb_search_num_live_docs"]["foo2"], 101);
+      assertEqual(metrics["arangodb_search_num_live_docs"]["foo3"], 102);
+
+      assertEqual(metrics["arangodb_search_num_segments"]["foo1"], 9);
+      assertEqual(metrics["arangodb_search_num_segments"]["foo2"], 12);
+      assertEqual(metrics["arangodb_search_num_segments"]["foo3"], 1);
+
+      assertEqual(metrics["arangodb_search_num_files"]["foo1"], 54);
+      assertEqual(metrics["arangodb_search_num_files"]["foo2"], 72);
+      assertEqual(metrics["arangodb_search_num_files"]["foo3"], 6);
+
+      assertTrue(metrics["arangodb_search_index_size"]["foo1"] > 0);
+      assertTrue(metrics["arangodb_search_index_size"]["foo2"] > 0);
+      assertTrue(metrics["arangodb_search_index_size"]["foo3"] > 0);
+    }
+  };
+}
+
+////////////////////////////////////////////////////////////////////////////////
```

Comment:
can we also add a test checking json format?

Reply:
- [ ] Add json test